src/Tasks/Microsoft.NET.Build.Tasks/GenerateSupportedTargetFrameworkAlias.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Frameworks;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class GenerateSupportedTargetFrameworkAlias : TaskBase
+    {
+        [Required]
+        public ITaskItem[] SupportedTargetFramework { get; set; }
+
+        [Required]
+        public string TargetFrameworkMoniker { get; set; }
+
+        public string TargetPlatformMoniker { get; set; }
+
+        public bool UseWpf { get; set; }
+
+        public bool UseWindowsForms { get; set; }
+
+        [Output]
+        public ITaskItem[] SupportedTargetFrameworkAlias { get; private set; }
+
+        protected override void ExecuteCore()
+        {
+            var targetFramework = NuGetFramework.Parse(TargetFrameworkMoniker);
+            if (!(targetFramework.Framework.Equals(".NETCoreApp") && targetFramework.Version >= new Version(5, 0)) && !UseWpf && !UseWindowsForms)
+            {
+                // Target platform properties were defaulted to windows prior to 5.0, ignore these values
+                TargetPlatformMoniker = string.Empty;
+            }
+
+            IList<ITaskItem> convertedTfms = new List<ITaskItem>();
+            foreach (var tfm in SupportedTargetFramework)
+            {
+                var targetFrameworkAlias = NuGetFramework.ParseComponents(tfm.ItemSpec, TargetPlatformMoniker).GetShortFolderName();
+                var newItem = new TaskItem(tfm);
```

Comment:
This causes it to inherit the metadata?

Reply:
Yes, this copies the item including the metadata.